### PR TITLE
fix: test inverse MCA differently depending on architecture

### LINKED ISSUE #72: Fix inverse transform changing the distribution a lot for MCA
The current state of `inverse_transform` during mca, yields in the test [`test_inverse_from_coord_mca`](https://github.com/octopize/saiph/blob/fe471cd47fd0b03ced8d30c7ca5408d8be3bcf06/saiph/inverse_transform_test.py#L114), that there is a (big) discrepancy of results between MacOS M1 (the `right` result) and Linux Intel (the `left` result).
Indeed, the test passes for MacOS, but fails on Linux, with the following values.

```text
saiph/inverse_transform_test.py:136: in test_inverse_from_coord_mca
    assert_series_equal(
pandas/_libs/testing.pyx:52: in pandas._libs.testing.assert_almost_equal
    ???
pandas/_libs/testing.pyx:167: in pandas._libs.testing.assert_almost_equal
    ???
E   AssertionError: Series are different
E   
E   Series values are different (75.0 %)
E   [index]: [count, mean, std, min, 25%, 50%, 75%, max]
E   [left]:  [683.0, 1.16398243045388, 0.5648994509823707, 1.0, 1.0, 1.0, 1.0, 4.0]
E   [right]: [683.0, 4.430453879941435, 2.74087303323353, 1.0, 2.0, 4.0, 5.5, 10.0]
```

Notice how the mean and the std are quite different from the intended results. Note that this is with `use_max_modalities=False`, which means we are using some randomness to pick the modalities for the categories.

What's interesting is that this test has values that are quite ok when we go back to a previous commit, 091ffdf.
This is no coincidence, as 18afa43 , the commit after 091ffdf,  introduced the concept of randomization.

We can reproduce this by setting `use_max_modalities=True`, because this uses no randomization. Using this,
we get values that are closer to the MacOS M1 values, but not exactly, which given that there is a seed set, should be the same . And it's even stranger that two Linux machines with Intel processors do not have these the same.

@jpetot  and I traced it back to the output of the SVD beeing (quite a bit) different on our respective architectures (we did not compare both Linux machines).

We looked around, and couldn't really find a cause to why the SVD was computing something different for both architectures, given that they have the same inputs (we checked). This [numpy issue](https://github.com/numpy/numpy/issues/21313) seems to say that having a matrix that's close to singular might lead to different eigenvectors being produced, but there is not a lot of other opinions out there (we haven't check thoroughly)

Here below you'll have the comparison of the distributions for every variable using `use_max_modalities=True`
[combined.txt](https://github.com/octopize/saiph/files/8772412/combined.txt)

### PR DESCRIPTION
**Pull request by tcrasset**
I updated the GitHub actions workflow to run on a MacOS machine too.

EDIT: I changed the `fail-fast` behavior 

I specified `fail-fast : false`, which means that if one of the runs fails, the other one will NOT be stopped. Setting it to `true`(default) would allow us to not waste some CI minutes, which we have a limited supply of (:dollar: :dollar: :dollar: ).

A use for `fail-fast: False` would be that we are able to know that a test failed on one machine and not the other, without it, we wouldn't be able to know without re-running, kind of defeating the purpose of having two tests. In addition, if they both start at the same time (more or less) (they run in parallel), most of the time is spent in setting up the dependencies, so these minutes will be lost either way. 

Let me know what you think.

- feat: use files generated on same machine to test inverse_transform
- feat: run github workflows on a MacOS machine

Closes #72

### CODE REVIEW

File: saiph/inverse_transform_test.py [hunk 1]
```diff
@@ -128,28 +164,61 @@ def test_inverse_from_coord_mca(
...
+    to_compare = [
+        CompareDistributions(label="count"),
+        CompareDistributions(label="mean", atol=0.25),
+        CompareDistributions(label="std", atol=0.35),
+        # assert equal for the min as there are many low values
+        CompareDistributions(label="min"),
+        CompareDistributions(label="max"),
+        CompareDistributions(label="25%", atol=1),
+        CompareDistributions(label="50%", atol=1),
+        CompareDistributions(label="75%", atol=2),
```
Comment (jpetot) [inline saiph/inverse_transform_test.py:182]: Maybe you can have a df or a dictionary to manage all absolute tolerance of the two tests at the same place.

**Review by jpetot — APPROVED** [review]
I am really happy that you fixed it, I recommend you two save all ` atol` in an object like a dictionary

**Comment by tcrasset** [comment]
It seems like the tests have different behavior on our local machines than on the machines that GitHub actions uses... This makes the problem harder ...

**Comment by ykatchou** [comment]
As this PR did not move for 18 months, i suggest to close it.

**Pull request closed without merging**